src/util.js

```diff
@@ -55,26 +55,55 @@ export function parseFunction (str) {
 
 	str = str.trim();
 
-	const isFunctionRegex = /^([a-z]+)\((.+?)\)$/i;
-	const isNumberRegex = /^-?[\d.]+$/;
+	const isFunctionRegex = /^(?<name>[a-z-]+)\((?<args>.+?)\)$/i;
+	const isNumberRegex = /^[+-]?\d?\.?\d+(e[+-]?\d+)?$/i;
 	let parts = str.match(isFunctionRegex);
 
 	if (parts) {
 		// It is a function, parse args
 		let args = [];
-		parts[2].replace(/\/?\s*([-\w.]+(?:%|deg)?)/g, ($0, arg) => {
+		parts.groups.args.replace(/\/?\s*([+-\w.e]+(?:%|deg|g?rad|°|pi|turn)?)/gi, ($0, arg) => {
 			if (/%$/.test(arg)) {
 				// Convert percentages to 0-1 numbers
 				arg = new Number(arg.slice(0, -1) / 100);
 				arg.type = "<percentage>";
 			}
 			else if (/deg$/.test(arg)) {
 				// Drop deg from degrees and convert to number
-				// TODO handle other units too
 				arg = new Number(+arg.slice(0, -3));
 				arg.type = "<angle>";
 				arg.unit = "deg";
 			}
+			else if (/°$/.test(arg)) { // this notation is not valid in CSS
+				// Drop degree sign from degrees and convert to number
+				arg = new Number(+arg.slice(0, -1));
+				arg.type = "<angle>";
+				arg.unit = "deg";
+			}
+			else if (/grad$/.test(arg)) { // check for 'grad' needs to come before check for 'rad' 
```

Comment:
This repeated regex testing is inefficient. You can have a regex that just reads the unit into a group and then does simple string comparison. This will also help with the repetitive code.

Reply:
I agree, just didn’t want to change too much.